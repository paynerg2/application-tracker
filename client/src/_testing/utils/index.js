import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../../_reducers';
import { middlewares } from '../../app/store';

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};
