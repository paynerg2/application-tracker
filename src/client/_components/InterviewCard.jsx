import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { interviewActions } from '../_actions';

const CardListItem = styled.li`
    /*Adust with media query */
    width: 30%;
    height: 25vh;
    padding: 15px;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 13px;
    background-color: #fff;
    list-style-type: none;
    margin: 10px;
    display: grid;
    grid-template-columns: 66% auto;
    box-shadow: 1px 2px 6px 1px lightgrey;
    border-radius: 2%;

    &:hover {
        box-shadow: 1px 2px 6px 1px lightblue;
    }
`;

const Button = styled.button`
    border: none;
    padding: 4px 7px 4px 7px;
    background: #ff8500;
    color: #fff;
    margin-top: 5px;
    box-shadow: 1px 1px 4px #dadada;
    -moz-box-shadow: 1px 1px 4px #dadada;
    -webkit-box-shadow: 1px 1px 4px #dadada;
    border-radius: 3px;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    font-family: 8px Arial, Helvetica, sans-serif;

    &:hover {
        background: #ea7b00;
        color: #fff;
    }
`;

class InterviewCard extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isSelected: false
        };

        this.handleDelete = this.handleDelete.bind(this);
    }

    handleDelete(id) {
        const { dispatch } = this.props;
        dispatch(interviewActions.delete(this.props.interview._id));
    }

    renderCard = interview => {
        const { startTime, location, _id, applicationId } = interview;
        return (
            <React.Fragment>
                <div>{startTime}</div>
                <div>{location}</div>

                {this.state.isSelected && (
                    <div>
                        <Link
                            to={`applications/${applicationId}/interviews/${_id}`}
                        >
                            <Button>edit</Button>
                        </Link>
                        <Button onClick={this.handleDelete}>Delete</Button>
                    </div>
                )}
            </React.Fragment>
        );
    };

    render() {
        return (
            <CardListItem
                onMouseEnter={() => this.setState({ isSelected: true })}
                onMouseLeave={() => this.setState({ isSelected: false })}
            >
                {this.renderCard(this.props.interview)}
            </CardListItem>
        );
    }
}

const connectedInterviewCard = connect(null)(InterviewCard);
export { connectedInterviewCard as InterviewCard };
