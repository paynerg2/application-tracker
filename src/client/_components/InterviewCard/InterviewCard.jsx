import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { interviewActions } from '../../_actions';
import { CardListItem, Button } from '../Card.styles';

class InterviewCard extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isSelected: false
        };

        this.handleDelete = this.handleDelete.bind(this);
    }

    handleDelete(id) {
        const { dispatch } = this.props;
        dispatch(interviewActions.delete(this.props.interview._id));
    }

    renderCard = interview => {
        const { startTime, location, _id, applicationId } = interview;
        return (
            <React.Fragment>
                <div>{startTime}</div>
                <div>{location}</div>

                {this.state.isSelected && (
                    <div>
                        <Link
                            to={`applications/${applicationId}/interviews/${_id}`}
                        >
                            <Button>edit</Button>
                        </Link>
                        <Button onClick={this.handleDelete}>Delete</Button>
                    </div>
                )}
            </React.Fragment>
        );
    };

    render() {
        const { interview } = this.props;
        if (!interview) {
            return null;
        }
        return (
            <CardListItem
                onMouseEnter={() => this.setState({ isSelected: true })}
                onMouseLeave={() => this.setState({ isSelected: false })}
            >
                {this.renderCard(interview)}
            </CardListItem>
        );
    }
}

InterviewCard.propTypes = {
    interview: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired
};

const connectedInterviewCard = connect(null)(InterviewCard);
export { connectedInterviewCard as InterviewCard };
