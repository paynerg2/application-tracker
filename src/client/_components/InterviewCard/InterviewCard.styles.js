import styled from 'styled-components';

export const DateDisplay = styled.div`
    display: flex;
    flex-direction: column;
    width: 90px;
    height: 115px;
    margin: 4px;
`;
DateDisplay.displayName = 'DateDisplay';

export const DateNumber = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: #fff;
    color: #1995ad;
    width: 100%;
    height: 50%;
    font-size: 25px;
    text-align: center;
`;
DateNumber.displayName = 'DateNumber';

export const Month = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: #1995ad;
    color: #fff;
    text-align: center;
    height: 16.6%;
    width: 100%;
`;
Month.displayName = 'Month';

export const Time = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: #1995ad;
    color: #fff;
    text-align: center;
    height: 16.6%;
    width: 100%;
`;
Time.displayName = 'Time';
