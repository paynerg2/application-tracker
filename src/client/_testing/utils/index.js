import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../../_reducers';

export const checkProps = (component, expectedProps) => {
    const propsError = checkPropTypes(
        component.propTypes,
        expectedProps,
        'props',
        component.name
    );
    return propsError;
};
