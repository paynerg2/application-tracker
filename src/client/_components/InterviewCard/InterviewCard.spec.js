import React from 'react';
import { shallow } from 'enzyme';

import { InterviewCard } from './InterviewCard';
import { testStore } from '../../_testing/utils';

const setUp = (props = {}, initialState = {}) => {
    const store = testStore(initialState);
    const component = shallow(
        <InterviewCard store={store} {...props} />
    ).dive();
    return component;
};

describe('Interview Card Component', () => {
    let mockFunction, wrapper;
    beforeEach(() => {
        mockFunction = jest.fn();
        const props = {
            dispatch: mockFunction,
            interview: {
                startTime: '5:55pm',
                location: 'test',
                _id: 'test',
                applicationId: 'test'
            }
        };
        wrapper = setUp(props);
    });

    it('renders without crashing', () => {
        expect(wrapper).toBeDefined();
    });

    describe('when the component is selected', () => {
        beforeEach(() => {
            const cardListItemWrapper = wrapper.find('CardListItem');
            cardListItemWrapper.simulate('mouseEnter');
        });

        it('shows additional buttons', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when the component is not selected', () => {
        it('does not show additional buttons', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('returns to unselected state on mouse leave event', () => {
            const CardListItemWrapper = wrapper.find('CardListItem');
            CardListItemWrapper.simulate('mouseEnter');
            CardListItemWrapper.simulate('mouseLeave');
            expect(wrapper).toMatchSnapshot();
        });
    });
});
