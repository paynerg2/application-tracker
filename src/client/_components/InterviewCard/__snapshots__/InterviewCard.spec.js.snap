// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interview Card Component when the component is not selected does not show additional buttons 1`] = `
<CardListItem
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    5:55pm
  </div>
  <div>
    test
  </div>
</CardListItem>
`;

exports[`Interview Card Component when the component is not selected returns to unselected state on mouse leave event 1`] = `
<CardListItem
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    5:55pm
  </div>
  <div>
    test
  </div>
</CardListItem>
`;

exports[`Interview Card Component when the component is selected shows additional buttons 1`] = `
<CardListItem
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    5:55pm
  </div>
  <div>
    test
  </div>
  <div>
    <Link
      replace={false}
      to="applications/test/interviews/test"
    >
      <Button>
        edit
      </Button>
    </Link>
    <Button
      onClick={[Function]}
    >
      Delete
    </Button>
  </div>
</CardListItem>
`;
